lib/src/lib.rs

```diff
@@ -660,6 +676,10 @@ impl Repo {
         let repo = Repo {
             path: repo_path.to_owned(),
             pub_key: pk,
+            write_stats: Arc::new(Mutex::new(WriteStats {
```

Comment:
Well, it have faster performance. 

Ideally, no synchronization would be needed. Seems to me there's just one thread that is incrementing these values. It might as well have another channel and send results back when finished or something like that.